### PR TITLE
ButtonLink component fix props and styles

### PR DESCRIPTION
**Pull request by janicekt**
Snímek obrazovky 2022-03-08 v 10 31 34
Fix bad props in ButtonLink and define text-decoration none to prevent default styles...

### CODE REVIEW

File: packages/web-react/src/components/Button/useButtonAriaProps.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ export function useButtonAriaProps(props: AriaButtonProps<ElementType>): ButtonA
       role: 'button',
       href: elementType === 'a' && disabled ? undefined : href,
       target: elementType === 'a' ? target : undefined,
-      type,
+      type: elementType === 'a' && type === 'button' ? undefined : type,
```
Comment (literat) [inline packages/web-react/src/components/Button/useButtonAriaProps.tsx:29]: I think that this should be enough.

```suggestion
      type: elementType === 'button'  ? type : undefined,
```

File: packages/web/src/components/Button/_Button.scss [hunk 2]
```diff
@@ -11,6 +11,7 @@
     align-items: center;
     padding: theme.$padding-y theme.$padding-x;
     text-align: center;
+    text-decoration: none;
```
Comment (literat) [inline packages/web/src/components/Button/_Button.scss:14]: Should this be here? Maybe having a `.Button_link` class will be better. What do you think @crishpeen?

**Review by literat — CHANGES_REQUESTED** [review]

**Comment by literat** [comment]
@janicekt Please also add a test for this fix. That there shouldn't be `type` attribute on `a` tag in `ButtonLink` component. Thanks.

File: packages/web-react/src/components/Button/useButtonAriaProps.tsx (hunk 1 above)
Comment (janicekt) [reply packages/web-react/src/components/Button/useButtonAriaProps.tsx:29]: in this condition there will never be an elementType button. I'm just trying not to set the default property type "button" here, which is set here at the top. Anchor has type attribute but it is used to define the mime type of the target document.

Comment (janicekt) [reply packages/web-react/src/components/Button/useButtonAriaProps.tsx:29]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-type

File: packages/web/src/components/Button/_Button.scss (hunk 2 above)
Comment (janicekt) [reply packages/web/src/components/Button/_Button.scss:14]: although it should rather be a modifier `Button--link`, but in this case it seemed useless to me. It is only always set that the text in the button will not be underlined, which will always be the case.

Comment (crishpeen) [reply packages/web/src/components/Button/_Button.scss:14]: I would keep it here as suggested in your PR. It's easier to use and we do not differ if you use `<button>` or `<a>` in button implementation in the web package.

**Review by crishpeen — APPROVED** [review]

**Comment by janicekt** [comment]
> @janicekt Please also add a test for this fix. That there shouldn't be `type` attribute on `a` tag in `ButtonLink` component. Thanks.

done

**Review by literat — APPROVED** [review]

**Pull request merged**